File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.cpp
class Solution {
public:
    int maxCoins(vector<int>& piles) {
        int totalCoins = 0, chance = 0, totalChance = piles.size()/3;
        sort(piles.begin(), piles.end());
        for(int i=piles.size()-2; chance<totalChance; i-=2, chance++)
            totalCoins += piles[i];
        return totalCoins;
    }
};

File: 1561-maximum-number-of-coins-you-can-get/1561-maximum-number-of-coins-you-can-get.java
class Solution {
    public int maxCoins(int[] piles) {
        int totalCoins = 0, chance = 0, totalChance = piles.length/3;
        Arrays.sort(piles);
        for(int i=piles.length-2; chance<totalChance; i-=2, chance++)
            totalCoins += piles[i];
        return totalCoins;
    }
}